packages/core/src/Stores/client-store.js

```diff
@@ -1910,6 +1910,13 @@ export default class ClientStore extends BaseStore {
             if (this.accounts[obj_balance.loginid].is_virtual) {
                 this.root_store.notifications.resetVirtualBalanceNotification(obj_balance.loginid);
             }
+
+            //temporary workaround to sync this.loginid with selected wallet loginid
```

Comment:
temporary workarounds typically stays forever :D question, why that feature is even needed? maybe it would be possible to fix it by removing whatever introduced this bug?

Reply:
hey @wojciech-deriv 
unfortunately we can't remove something that doesn't exist in terms of UI
The reason behind this bug is that currently our mobx state and reactquery state not in sync.
To fix this bug, we need to properly synchronise our wallets package with external core package.
Now. when you switch between the wallets in TH, legacy deriv-app doesn't know what wallet was selected, and in our case, when we updating balance after transfer/withdraw we are using legacy setBalance with legacy loginid related to the one we first logged in with. The proper one we have in localstorage (every time we switching between the wallets we adding loginid to localstorage). That's why for now i'm using this workaround.